dashboard/app/controllers/api/v1/user_school_infos_controller.rb

```diff
@@ -0,0 +1,58 @@
+class Api::V1::UserSchoolInfosController < ApplicationController
+  before_action :authenticate_user!
+  before_action :load_user_school_info
+  load_and_authorize_resource
+
+  # PATCH /api/v1/users_school_infos/<id>/update_last_confirmation_date
+  def update_last_confirmation_date
+    @user_school_info.update!(last_confirmation_date: DateTime.now)
+
+    head :no_content
+  end
+
+  # PATCH /api/v1/users_school_infos/<id>/update_end_date
+  def update_end_date
+    ActiveRecord::Base.transaction do
+      @user_school_info.update!(end_date: DateTime.now, last_confirmation_date: DateTime.now)
+      @user_school_info.user.update!(properties: {last_seen_school_info_interstitial: DateTime.now})
+
+      head :no_content
```

Comment:
this should be outside of the transaction

Reply:
Good catch! thanks